src/sensorkit.cs

```diff
@@ -982,4 +1076,309 @@ interface SRSupplementalCategory : NSCopying, NSSecureCoding {
+
+		[Export ("value", ArgumentSemantic.Strong)]
+		NSMeasurement<NSUnitElectricPotentialDifference> Value { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (17, 4), MacCatalyst (17, 4)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SRElectrocardiogramSample : NSCopying, NSSecureCoding {
+
+		[Export ("date", ArgumentSemantic.Strong)]
+		NSDate Date { get; }
+
+		[Export ("frequency", ArgumentSemantic.Strong)]
+		NSMeasurement<NSUnitFrequency> Frequency { get; }
+
+		[Export ("session", ArgumentSemantic.Strong)]
+		SRElectrocardiogramSession Session { get; }
+
+		[Export ("lead", ArgumentSemantic.Assign)]
+		SRElectrocardiogramLead Lead { get; }
+
+		[Export ("data", ArgumentSemantic.Copy)]
+		SRElectrocardiogramData [] Data { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (17, 4), MacCatalyst (17, 4)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SRElectrocardiogramSession : NSCopying, NSSecureCoding {
+
+		[Export ("state", ArgumentSemantic.Assign)]
+		SRElectrocardiogramSessionState State { get; }
+
+		[Export ("sessionGuidance", ArgumentSemantic.Assign)]
+		SRElectrocardiogramSessionGuidance SessionGuidance { get; }
+
+		[Export ("identifier")]
+		string Identifier { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (17, 4), MacCatalyst (17, 4)]
+	public enum SRPhotoplethysmogramOpticalSampleCondition {
+		[Field ("SRPhotoplethysmogramOpticalSampleConditionSignalSaturation")]
+		SignalSaturation,
+		[Field ("SRPhotoplethysmogramOpticalSampleConditionUnreliableNoise")]
+		UnreliableNoise,
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (17, 4), MacCatalyst (17, 4)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SRPhotoplethysmogramOpticalSample : NSCopying, NSSecureCoding {
+
+		[Export ("emitter")]
+		nint Emitter { get; }
+
+		[Export ("activePhotodiodeIndexes", ArgumentSemantic.Strong)]
+		NSIndexSet ActivePhotodiodeIndexes { get; }
+
+		[Export ("signalIdentifier")]
+		nint SignalIdentifier { get; }
+
+		[Export ("nominalWavelength", ArgumentSemantic.Strong)]
+		NSMeasurement<NSUnitLength> NominalWavelength { get; }
+
+		[Export ("effectiveWavelength", ArgumentSemantic.Strong)]
+		NSMeasurement<NSUnitLength> EffectiveWavelength { get; }
+
+		[Export ("samplingFrequency", ArgumentSemantic.Strong)]
+		NSMeasurement<NSUnitFrequency> SamplingFrequency { get; }
+
+		[Export ("nanosecondsSinceStart")]
+		long NanosecondsSinceStart { get; }
+
+		[NullAllowed]
+		[Export ("normalizedReflectance", ArgumentSemantic.Strong)]
+		NSNumber NormalizedReflectance { get; }
+
+		[NullAllowed]
+		[Export ("whiteNoise", ArgumentSemantic.Strong)]
+		NSNumber WhiteNoise { get; }
+
+		[NullAllowed]
+		[Export ("pinkNoise", ArgumentSemantic.Strong)]
+		NSNumber PinkNoise { get; }
+
+		[NullAllowed]
+		[Export ("backgroundNoise", ArgumentSemantic.Strong)]
+		NSNumber BackgroundNoise { get; }
+
+		[NullAllowed]
+		[Export ("backgroundNoiseOffset", ArgumentSemantic.Strong)]
+		NSNumber BackgroundNoiseOffset { get; }
```

Comment:
I think we can BindAs these members, the underlying type seems either documented in ObjC or explicit in the Swift API.